packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -325,27 +309,70 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
         // of the LP fees, the relayer fees and the instant relay fee.
         uint256 feesTotal =
             _getAmountFromPct(
-                relay.realizedLpFeePct + depositData.slowRelayFeePct + depositData.instantRelayFeePct,
+                relayData.realizedLpFeePct + depositData.slowRelayFeePct + depositData.instantRelayFeePct,
                 depositData.amount
             );
 
         l1Token.safeTransferFrom(msg.sender, depositData.l1Recipient, depositData.amount - feesTotal);
 
-        emit RelaySpedUp(depositHash, msg.sender, relay.realizedLpFeePct);
+        emit RelaySpedUp(depositHash, msg.sender, relayData);
     }
 
     /**
      * @notice Reward relayers if a pending relay price request has a price available on the OptimisticOracle. Mark
      * the relay as complete.
      * @param depositData Unique set of L2 deposit data that caller is trying to settle a relay for.
+     * @param relayData Parameters of Relay that caller is attempting to settle. Must hash to the stored relay hash
+     * for this deposit.
+     * @param ancillaryData Relay price request's ancillary data. Used to check status of relay price request.
+     * @param request Relay price request struct. Used to check status of relay price request.
      */
-    function settleRelay(DepositData memory depositData) public nonReentrant() {
+    function settleRelay(
+        DepositData memory depositData,
+        RelayData memory relayData,
+        bytes memory ancillaryData,
+        SkinnyOptimisticOracleInterface.Request memory request
```

Comment:
Have fun calling this method

Reply:
A lot of the information in request, we might be able to infer from relay and deposit data, so (in a follow-up) we may be able to get away with having the user only providing some of these fields.

Reply:
ye... this is going to be painful. ideally re-constructed in the bot so no on-chain calls